### ISSUE TITLE
 tpm2_verifysignature ignores the -k option

### ISSUE THREAD

**Issue by danglus** [01org/tpm2.0-tools#265]
please see below:
-----------
root@frc-nuc:~/tpm/test# tpm2_verifysignature  -k  0x80000001  -s sign.out -m rnd.txt -T device
ERROR: --keyHandle (-k) or --keyContext (-c) and --sig (-s) and --ticket (-t) must be specified
------------

**Comment by williamcroberts** [comment]
This appears to be because the ticket file is not being specified. What exactly do you want changed? Do you want to make ticket file optional?

**Comment by williamcroberts** [comment]
@danglus can you please comment on what you want, I am getting ready to cut a new release and would like to include this.

**Comment by danglus** [comment]
@williamcroberts , sure. For some reason, the TPM2.0-TSS library and, even these tool, are not working well with the Infinion TPM that I have (see thread here: https://github.com/01org/TPM2.0-TSS/issues/354).  From what I can understand, the library failing to save context. Anyway, as a workaround, I have used the new "-T device" option, and avoided creating context files (which fails).

This way, I managed to create a primary key, create a key and use it to sign a file (always using handles instead of context files).  
However, in order to verify a signature, I wanted to give the command the key handle (using the -k option), but unfortunately, it failed. 

To sum it up, I wanted a way for tpm2_verifysignature to get the key handle instead of a key context file.

  Thanks,
Dan

**Comment by danglus** [comment]
@williamcroberts - one small clarification. I'm not sure that this is really so important. I opened this ticket, mainly, because there is an existing feature that doesn't work. In any case, the application that I need is beyond the tools current capabilities so, I'll have to write custom C++ code anyway.

**Comment by williamcroberts** [comment]
@danglus OK, but the body of the bug ticket shows that you didn't specify the ticket file (--ticket or -t) option, which is why you're getting the error. You need to specify that. The ticket file is an output file parameter.

Try this command:
```
# tpm2_verifysignature -T device -k 0x80000001 -s sign.out -m rnd.txt -t ticket.out
```

**Comment by williamcroberts** [comment]
Note that -T for the tcti context is not the same as -t for the ticket file.

**Comment by danglus** [comment]
I see now. Thanks!
